src/fastgltf.cpp

```diff
@@ -1303,366 +1313,445 @@ fg::Expected<fg::Asset> fg::Parser::parse(simdjson::dom::object root, Category c
 	return Expected(std::move(asset));
 }
 
-fg::Error fg::Parser::parseAccessors(simdjson::dom::array& accessors, Asset& asset) {
+fg::Error fg::Parser::parseAccessors(simdjson::ondemand::array& accessors, Asset& asset) {
     using namespace simdjson;
 
-	asset.accessors.reserve(accessors.size());
+	// asset.accessors.reserve(accessors.size());
     for (auto accessorValue : accessors) {
-        // Required fields: "componentType", "count"
-        Accessor accessor = {};
-        dom::object accessorObject;
+        ondemand::object accessorObject;
         if (accessorValue.get_object().get(accessorObject) != SUCCESS) {
             return Error::InvalidGltf;
         }
 
-        std::uint64_t componentType;
-        if (accessorObject["componentType"].get_uint64().get(componentType) != SUCCESS) {
-            return Error::InvalidGltf;
-        }
-		accessor.componentType = getComponentType(static_cast<std::underlying_type_t<ComponentType>>(componentType));
-        if (accessor.componentType == ComponentType::Double && !hasBit(options, Options::AllowDouble)) {
-            return Error::InvalidGltf;
-        }
-
-        std::string_view accessorType;
-        if (accessorObject["type"].get_string().get(accessorType) != SUCCESS) {
-            return Error::InvalidGltf;
-        }
-		accessor.type = getAccessorType(accessorType);
-
-        std::uint64_t accessorCount;
-        if (accessorObject["count"].get_uint64().get(accessorCount) != SUCCESS) {
-            return Error::InvalidGltf;
-        }
-		accessor.count = static_cast<std::size_t>(accessorCount);
-
-
-        std::uint64_t bufferView;
-        if (accessorObject["bufferView"].get_uint64().get(bufferView) == SUCCESS) {
-            accessor.bufferViewIndex = static_cast<std::size_t>(bufferView);
-        }
+        auto& accessor = asset.accessors.emplace_back();
+        accessor.byteOffset = 0UL;
+        accessor.normalized = false;
 
-        // byteOffset is optional, but defaults to 0
-        std::uint64_t byteOffset;
-        if (auto error = accessorObject["byteOffset"].get_uint64().get(byteOffset); error == SUCCESS) {
-            accessor.byteOffset = static_cast<std::size_t>(byteOffset);
-        } else if (error != NO_SUCH_FIELD) {
-            return Error::InvalidGltf;
-        }
+        // Function for parsing the min and max arrays for accessors
+        auto parseMinMax = [&](simdjson::ondemand::array& array, decltype(Accessor::max)& ref) -> fastgltf::Error {
+			return Error::None;
```

Comment:
So the first line of the function is a return?

Reply:
In my original comment I mentioned that I committed some things I didn't exactly intend. I'll update the branch in a second with a few fixes.